tasks/centos.yml

```diff
@@ -38,20 +38,31 @@
     - lv.1.create is defined
     - lv.1.create|bool
 
-- name: centos | checking for scsi devices
-  command: sg_scan
-  become: true
-  register: scsi_devices
-  changed_when: false
+- block:
+  - name: centos | installing sg3_utils
+    package:
+    name: sg3_utils
+    state: present
+    become: true
 
-- name: centos | rescanning for new disks
-  command: /usr/bin/rescan-scsi-bus.sh
-  become: true
-  changed_when: false
-  when: scsi_devices.stdout|length > 0
+  - name: centos | checking for scsi devices
+    command: sg_scan
+    become: true
+    register: scsi_devices
+    changed_when: false
 
-- name: centos | rescanning for resized disks
-  command: /usr/bin/rescan-scsi-bus.sh -s
-  become: true
-  changed_when: false
-  when: scsi_devices.stdout|length > 0
+  - name: centos | rescanning for new disks
+    command: /usr/bin/rescan-scsi-bus.sh
+    become: true
+    changed_when: false
+    when: scsi_devices.stdout|length > 0
+
+  - name: centos | rescanning for resized disks
+    command: /usr/bin/rescan-scsi-bus.sh -s
+    become: true
+    changed_when: false
+    when: scsi_devices.stdout|length > 0
+
+  when:
+    - "'sd' in ansible_devices.keys()"
+    - "'vd' in ansible_devices.keys()"
```

Comment:
This is doing an exact match, so we'll never hit this code.

Also, are we sure that these are the only device names we'll see? hdX is for IDE drives, if they still exist.